### PR TITLE
fix: unexpected return error

### LINKED ISSUE #8658: [Bug] Unexpected return error on empty mustache binding {{}}
image

### PR DESCRIPTION
**Pull request by eco-monk**
## Description

Handle empty new lines in bindings

Fixes #8658

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Test A:
- unit tests

Test B:
- enter only new line in any js field with {{}} binding 
```
{{

}}
```
- notice "Unexpected return error" doesn't show up anymore

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/unexpected-return-error 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.14 **(0)** | 37.09 **(0.02)** | 33.96 **(0)** | 55.64 **(0)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :green_circle: | app/client/src/workers/evaluate.ts | 88.17 **(0.26)** | 78.13 **(1.46)** | 77.78 **(0)** | 87.5 **(0.29)**</details>

### CODE REVIEW

File: app/client/src/workers/evaluate.test.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import {
   ENTITY_TYPE,
 } from "entities/DataTree/dataTreeFactory";
 import { RenderModes } from "constants/WidgetConstants";
+import { PropertyEvaluationErrorType } from "utils/DynamicBindingUtils";
```
Comment (hetunandu) [inline app/client/src/workers/evaluate.test.ts:8]:
```suggestion
```

**Review by hetunandu — APPROVED** [review]

File: app/client/src/workers/evaluate.ts [hunk 2]
```diff
@@ -145,12 +145,17 @@ export default function evaluate(
   evalArguments?: Array<any>,
   isTriggerBased = false,
 ): EvalResult {
+  let script = "",
+    scriptToLint = "";
```
Comment (hetunandu) [inline app/client/src/workers/evaluate.ts:149]:
```suggestion
  let script = "";
  let scriptToLint = "";
```

File: app/client/src/workers/evaluate.ts [hunk 3]
```diff
@@ -145,12 +145,17 @@ export default function evaluate(
...
   const unescapedJS = unEscapeScript(js);
   const scriptType = getScriptType(evalArguments, isTriggerBased);
-  const script = getScriptToEval(unescapedJS, scriptType);
-  // We are linting original js binding,
-  // This will make sure that the character count is not messed up when we do unescapejs
-  const scriptToLint = getScriptToEval(js, scriptType);
+
+  if (unescapedJS.length > 0) {
```
Comment (hetunandu) [inline app/client/src/workers/evaluate.ts:153]: Explain in comments why this if statement is needed

Comment (hetunandu) [reply app/client/src/workers/evaluate.ts:153]: If nothing is present to evaluate, lets just return back instead of evaluating

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/workers/evaluate.ts (hunk 3 above)
Comment (eco-monk) [reply app/client/src/workers/evaluate.ts]: Returning an empty response now
```
{
      errors: [],
      result: undefined,
      triggers: [],
};
```
---

Tried just `undefined`. Had to add null check in too many other places, as a consequence.

This is simpler.

File: app/client/src/workers/evaluate.ts [hunk 4]
```diff
@@ -145,12 +145,25 @@ export default function evaluate(
...
   isTriggerBased = false,
 ): EvalResult {
+  let script = "";
+  let scriptToLint = "";
   const unescapedJS = unEscapeScript(js);
   const scriptType = getScriptType(evalArguments, isTriggerBased);
-  const script = getScriptToEval(unescapedJS, scriptType);
+
+  // If nothing is present to evaluate, return back instead of evaluating
+  if (!(unescapedJS.length > 0)) {
```
Comment (hetunandu) [inline app/client/src/workers/evaluate.ts:154]:
```suggestion
  if (!unescapedJS.length) {
```

**Review by hetunandu — DISMISSED** [review]

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**